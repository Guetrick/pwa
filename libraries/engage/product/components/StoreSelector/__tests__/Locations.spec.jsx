import React from 'react';
import { shallow, mount } from 'enzyme';
import Locations from '../Locations';
import StoreSelectorContext from '../context';

const locations = [
  {
    code: 1,
    name: 'test1',
  },
  {
    code: 2,
    name: 'test2',
  },
];

describe('<Locations />', () => {
  it('should not render if no locations have been passed', () => {
    const wrapper = shallow((
      <StoreSelectorContext.Provider>
        <Locations />
      </StoreSelectorContext.Provider>
    ));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance()).toEqual(null);
  });

  it('should render locations', () => {
    const wrapper = mount((
      <StoreSelectorContext.Provider value={locations}>
        <Locations />
      </StoreSelectorContext.Provider>
    ));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Locations').getElements().length).toEqual(1);
  });
});
