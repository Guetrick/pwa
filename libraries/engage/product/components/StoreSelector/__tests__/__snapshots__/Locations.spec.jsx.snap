// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Locations /> should not render if no locations have been passed 1`] = `<Locations />`;

exports[`<Locations /> should render locations 1`] = `<Locations />`;
